### PR TITLE
Add the homepage products API to Spree Client

### PR DESCRIPTION
**Pull request by tmtrademarked**
This PR adds support for the homepage products API to the Spree Client. This will enable us to use this API in ba.com.

[PLAT-968](https://jira.blueapron.com/browse/PLAT-968)

### CODE REVIEW

File: lib/blue_apron/spree_client.rb [hunk 1]
```diff
@@ -133,6 +133,16 @@ def get_products(options = {})
...
     end
 
+    def get_homepage_products
+      response = get "/api/products/homepage"
+      response.keys.each do |key|
+        response[key].each do |product|
+          sanitize_product!(product)
+        end
+      end
+      response
```
Comment (dkgramming) [inline lib/blue_apron/spree_client.rb]: non-blocking: this hash iteration could be simplified a bit
```ruby
response.each do |_key, product|
  sanitize_product!(product)
end
```

Comment (tmtrademarked) [reply lib/blue_apron/spree_client.rb]: That doesn't quite work - each key in the response isn't a product, but rather a list of products. But the following would work, I believe:

```
response.each do |_key, products|
  products.each { |product| sanitize_product!(product) }
end
```

Thanks for the suggestion - much cleaner this way!

Comment (ramimassoud) [reply lib/blue_apron/spree_client.rb]: Definitely fine as is, but I think you can also do
```ruby 
response.values.flatten.each do |product|
  sanitize_product!(product)
end
```

**Comment by ramimassoud** [comment]
btw it might be worth it to bump the ruby version to whatever is running on BAcom now (2.5.??)
https://github.com/blueapron/spree-client-ruby/blob/336205a931eaf78c5637898896dcd79b8000e042/.ruby-version

**Pull request merged**